src/libstd/sys/unix/thread.rs

```diff
@@ -125,8 +126,10 @@ impl Thread {
     }
 
     pub fn sleep(dur: Duration) {
+        let secs = dur.as_secs();
+        assert!(secs <= (u64::MAX >> 1), "secs number is too big");
```

Comment:
The error happens because “The rqtp argument specified a nanosecond value less than zero or greater than or equal to 1000 million.” `u64::MAX >> 1` is not 1000 million.